src/components/LeftNav.js

```diff
@@ -141,10 +141,29 @@ class LeftNav extends Component<Props, State> {
   }
 
   addFolder = () => {
+    const folders = Object.keys(this.props.getFolders())
+    const regex = /new folder(\d*)/
+    let nextFolderNumber = 0
+    folders.forEach(folderName => {
+      const maxFolderNumber = folderName.match(regex)
+      if (maxFolderNumber) {
+        if (maxFolderNumber[1]) {
+          nextFolderNumber = Math.max(
+            parseInt(maxFolderNumber[1]) + 1,
+            nextFolderNumber,
+          )
+        } else {
+          nextFolderNumber = Math.max(1, nextFolderNumber)
+        }
+      }
+    })
```

Comment:
Ha, I feel like I'm on some sort of "we need more functions!" drive, but bear with me... what if we make a simple utility function that takes a string (presumably with one or more digits at the end) and returns a string, with the incremented number at the end? Then `addFolder` could take care of just the state change and I think it would be clearer what it's doing.

Reply:
No haha, I agree, that's really helpful feedback.
I don't think I did exactly what you were thinking.. but I personally think it helps/ makes sense for what we're doing. lmk.